cpp/src/unary/math_ops.cu

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "unary_ops.cuh"
+
+#include "utilities/type_dispatcher.hpp"
+
+#include <cmath>
+#include <algorithm>
+#include <type_traits>
+
+// trig functions
+
+struct DeviceSin {
+    template<typename T>
+    __device__
+    T apply(T data) {
+        return std::sin(data);
```

Comment:
Does `std::sin` etc. work in `__device__` code? CUDA has its own version of all of these functions, but I don't think they are `std`.

That reminds me, we might (down the road) want to provide a "fastmath" option which calls `__sin` and friends...

Reply:
It did work because there are [tests](https://github.com/rapidsai/cudf/pull/1163/files?file-filters%5B%5D=.py#diff-865b0f52b469d23a3783a48977756c9cL126) to check this. I didn't benchmark them especially because I don't know if all those math functions are even required. They aren't exposed in cuDF either.

Reply:
They should be exposed in cuDF. I thought part of this PR was to expose all of the unary ops in cuDF?  Weird that std::sin() works in a __device__ function.